cmd/kubeadm/app/util/kubeconfig_test.go

```diff
@@ -0,0 +1,174 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+)
+
+type configClient struct {
+	c  string
+	s  string
+	ca []byte
+}
+
+type configClientWithCerts struct {
+	c           *clientcmdapi.Config
+	clusterName string
+	userName    string
+	clientKey   []byte
+	clientCert  []byte
+}
+
+type configClientWithToken struct {
+	c           *clientcmdapi.Config
+	clusterName string
+	userName    string
+	token       string
+}
+
+func init() {
+	rand.Seed(time.Now().UTC().UnixNano())
+}
+
+func TestCreateBasicClientConfig(t *testing.T) {
+	var createBasicTest = []struct {
+		cc       configClient
+		expected string
+	}{
+		{configClient{}, ""},
+		{configClient{c: "kubernetes"}, ""},
+	}
+	for _, rt := range createBasicTest {
+		c := CreateBasicClientConfig(rt.cc.c, rt.cc.s, rt.cc.ca)
+		if c.Kind != rt.expected {
+			t.Errorf(
+				"failed CreateBasicClientConfig:\n\texpected: %d\n\t  actual: %d",
+				c.Kind,
+				rt.expected,
+			)
+		}
+	}
+}
+
+func TestMakeClientConfigWithCerts(t *testing.T) {
+	var createBasicTest = []struct {
+		cc          configClient
+		ccWithCerts configClientWithCerts
+		expected    string
+	}{
+		{configClient{}, configClientWithCerts{}, ""},
+		{configClient{c: "kubernetes"}, configClientWithCerts{}, ""},
+	}
+	for _, rt := range createBasicTest {
+		c := CreateBasicClientConfig(rt.cc.c, rt.cc.s, rt.cc.ca)
+		rt.ccWithCerts.c = c
+		cwc := MakeClientConfigWithCerts(
+			rt.ccWithCerts.c,
+			rt.ccWithCerts.clusterName,
+			rt.ccWithCerts.userName,
+			rt.ccWithCerts.clientKey,
+			rt.ccWithCerts.clientCert,
+		)
+		if cwc.Kind != rt.expected {
+			t.Errorf(
+				"failed MakeClientConfigWithCerts:\n\texpected: %d\n\t  actual: %d",
+				c.Kind,
+				rt.expected,
+			)
+		}
+	}
+}
+
+func TestMakeClientConfigWithToken(t *testing.T) {
+	var createBasicTest = []struct {
+		cc          configClient
+		ccWithToken configClientWithToken
+		expected    string
+	}{
+		{configClient{}, configClientWithToken{}, ""},
+		{configClient{c: "kubernetes"}, configClientWithToken{}, ""},
+	}
+	for _, rt := range createBasicTest {
+		c := CreateBasicClientConfig(rt.cc.c, rt.cc.s, rt.cc.ca)
+		rt.ccWithToken.c = c
+		cwc := MakeClientConfigWithToken(
+			rt.ccWithToken.c,
+			rt.ccWithToken.clusterName,
+			rt.ccWithToken.userName,
+			rt.ccWithToken.token,
+		)
+		if cwc.Kind != rt.expected {
+			t.Errorf(
+				"failed MakeClientConfigWithCerts:\n\texpected: %d\n\t  actual: %d",
+				c.Kind,
+				rt.expected,
+			)
+		}
+	}
+}
+
+func setEnvs() {
```

Comment:
There are a couple of problems here.  First, you can't assume that `/tmp` is a directory you can use.  Second, the random number may not be unique -- especially over time.  Third, you are cleaning up after yourself.

The pattern to use seems to be to use `ioutil.TempDir` and remote that directory after you are done.  See https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/clientcmd/loader_test.go#L128.

Finally -- it is unfortunate that you are having to set the environment variables.  I'd be afraid those would leak places you wouldn't expect.  It might make more sense to modify `GetEnvParams` so that you inject some values there.